File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.cpp
class Solution {
public:
    string processStr(string s) {
        string ans;
        for (char ch : s) {
            if (islower(ch)) {
                ans += ch;
            } else if (ch == '*') {
                if (!ans.empty()) {
                    ans.pop_back();
                }
            } else if (ch == '#') {
                ans += ans;
            } else {
                reverse(ans.begin(), ans.end());
            }
        }
        return ans;
    }
};

File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.py
class Solution:
    def processStr(self, s: str) -> str:
        ans = []

        for ch in s:
            if 'a' <= ch <= 'z':
                ans.append(ch)
            elif ch == '*':
                if ans:
                    ans.pop()
            elif ch == '#':
                ans += ans
            else:
                ans.reverse()

        return "".join(ans)